File: r/raindrops.R
raindrops <- function(number) {
  s <- ""
  if (number %% 3 == 0) {
    s <- paste(s, "Pling", sep = "")
  }
  if (number %% 5 == 0) {
    s <- paste(s, "Plang", sep = "")
  }
  if (number %% 7 == 0) {
    s <- paste(s, "Plong", sep = "")
  }
  if (s == "") {
    as.character(number)
  } else {
    s
  }
}

File: ocaml/raindrops.ml
let raindrop n =
  let f (m, s) = if n mod m = 0 then s else "" in
  let s =
    List.map f [ (3, "Pling"); (5, "Plang"); (7, "Plong") ] |> String.concat ""
  in
  match s with "" -> string_of_int n | _ -> s
